Engine/Src/Ancona/Core2D/Systems/PathFollowerSystem.cpp

```diff
@@ -0,0 +1,107 @@
+#include "PathFollowerSystem.hpp"
+
+using namespace ild;
+
+/* Component */
+PathFollowerComponent::PathFollowerComponent() {
+}
+
+void PathFollowerComponent::Update(float delta) {
+    if (_pathComponent->vertices().size() == 0) {
+        return;
+    }
+        
+    if (_action->Done()) {
+        StartNextPathSegment();
+    }
+}
+
+void PathFollowerComponent::StartNextPathSegment() {
+    auto lastVertexIndex = _nextVertexIndex;
+    _nextVertexIndex = NextVertexIndex();
+    auto time = NextSegmentTime(lastVertexIndex, _nextVertexIndex); 
+    
+    _action->
+        ResetAge()->
+        value(NextSegmentDistance())->
+        tween(time)->
+        duration(time);
+}
+
+int PathFollowerComponent::NextVertexIndex() {
+    auto nextVertexIndex = _nextVertexIndex;
+    nextVertexIndex += _followDirection;
+    
+    if (!IsVertexIndexValid(nextVertexIndex)) {
```

Comment:
Might be nit-picky, but You're basically checking to see if a vertex is out of bounds so you can bring it back in bounds. could we rename it to be IsVertexInBounds, then we can do if (IsVertexInBounds()) return vertex, then do special handling for when it is not?